include/tcnn/TextUtils.hpp

```diff
@@ -0,0 +1,69 @@
+///
+/// @par Text / character utilities
+///
+
+#ifndef TCNN_TEXT_UTILS_H
+#define TCNN_TEXT_UTILS_H
+
+#include <cstddef>
+#include <cstdint>
+#include <string_view>
+
+#include <strong_type/equality.hpp>
+#include <strong_type/ordered.hpp>
+#include <strong_type/strong_type.hpp>
+
+namespace tcnn {
+
+using image_id = strong::type<std::int8_t, struct image_id_tag, strong::ordered,
+                              strong::equality>;
+
+inline constexpr image_id TEXT_PAD_ID{0};
+inline constexpr image_id TEXT_VOCAB_SIZE{96}; // Number of printable ASCII
+                                               // characters (from space to ~)
+inline constexpr image_id TEXT_FIRST_PRINTABLE_CHAR{32};
+inline constexpr image_id TEXT_LAST_PRINTABLE_CHAR{126};
+inline constexpr image_id TEXT_PRINTABLE_RANGE_OFFSET{-31};
+
+struct TextGridParams {
+  std::string_view text;
```

Comment:
`string_view` is a lightweight container https://en.cppreference.com/w/cpp/string/basic_string_view.html

It allows us to store a safe view of the text without needing for us to find the end of the string ourselves.